### PR TITLE
Version support for warehouse schema gen

### PR DESCRIPTION
**Pull request by gane5hvarma**
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-transformer/158)

### CODE REVIEW

File: v0/destinations/bq/transform.js [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-function processSingleMessage(message, destination) {
-  return processWarehouseMessage("bq", message);
+function processSingleMessage(message, destination, schemaVersion) {
+  return processWarehouseMessage("bq", message, schemaVersion);
 }
 
 function process(event) {
-  return processSingleMessage(event.message, event.destination);
+  const schemaVersion = event.request.query.whSchemaVersion;
+  return processSingleMessage(event.message, event.destination, schemaVersion);
```
Comment (SumanthPuram) [inline v0/destinations/bq/transform.js:9]: We don't need event.destination as the 2nd argument.

Comment (psrikanth88) [reply v0/destinations/bq/transform.js:9]: removed

File: v0/destinations/util.js [hunk 2]
```diff
@@ -170,7 +170,32 @@ const getDataType = val => {
   return "string";
 };
 
-function safeTableName(provider, name = "") {
+const toSafeDBString = str => {
```
Comment (SumanthPuram) [inline v0/destinations/util.js:173]: There is a lot of warehouse specific code right now in this util. Might be a good time to move it out.


```
---warehouse/utils.js
  const v0 = require('v0/utils.js')
  const v1 = require('v1/utils.js')

  const getWarehouseUtil = function(schemaVersion) {
      if (schemaVersion == 'v0') {
         return v0;
      } if (schemaVersion == 'v1') {
         return v1;
      } else {
         // Default
         return v1;
      }
  }
```

```
---warehouse/v0/util.js
function safeTable() {
...
}
function safeColumn() {
...
}
function transformTableName() {
...
}
module.exports = {
safeTable,
safeColumn,
transformTableName
}
```
```
---warehouse/v1/util.js
function safeTable() {
...
}
function safeColumn() {
...
}
function transformTableName() {
...
}
module.exports = {
safeTable,
safeColumn,
transformTableName
}

 ```

File: warehouse/index.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const v0 = require("./v0/process.js");
+const v1 = require("./v1/process.js");
+
```
Comment (SumanthPuram) [inline warehouse/index.js:3]:
```
function getWarehouseUtil(schemaVersion) {
    case "v0":
      return {..defaultWHUtil, ...v0};
    case "v1":
      return v1;
    default:
      return v1;
}
```

```
function processWarehouseMessage(provider, message, schemaVersion) {
  const whUtil = getWarehouseImpl(schemaVersion)
  ...
  ...
  t = whUtil.safeTableName(a)
}
```
```
 --- warehouse/v0/util.js
 module.exports = {
  safeTableName,
  safe
}
```

**Pull request merged**